soroban-sdk/src/env.rs

```diff
@@ -549,6 +621,79 @@ impl Env {
         contract_id
     }
 
+    /// Checks if a top-level `require_auth` call has happened during the last
+    /// contract invocation for the given `address` and contract invocation.
+    ///
+    /// This also removes the record about the authorization, so calling this
+    /// for the second time with the same arguments will usually return `false`
+    /// (unless the identical authorization happened multiple times).
```

Comment:
The side effects of calling a "verify" function are surprising to me, and I think will be surprising to other people. Functions that verify things are usually idempotent. See below comment for a suggestion.

Reply:
That's a good suggestion, thanks. Replied there.